### PR TITLE
DP || Bellman Ford Algorithm to solve Single Source Shortest Path problem

### LINKED ISSUE #6915: DP || Bellman Ford Algorithm
### Description

A theory of how Bellman Ford Algorithm works using DP. 

### Domain

C/CPP

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by DwipshikhaLodh**
<hr>

## Description 📜

Explanation of how we can solve Single Source Shortest Path problem using Bellman Ford Algorithm which further uses DP approach.
Fixes #6915  

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [x] C/CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

File: C_CPP/Algorithmic Approaches/Dynamic/BellmanFordAlgorithm.md [hunk 1]
```diff
@@ -0,0 +1,560 @@
+# Bellman Ford Algorithm
+---
+---
```
Comment (kashishahuja2002) [inline C_CPP/Algorithmic Approaches/Dynamic/BellmanFordAlgorithm.md:3]: Remove these lines.

**Review by kashishahuja2002 — APPROVED** [review]
Great work.

**Review by kashishahuja2002 — CHANGES_REQUESTED** [review]
Add a relative link in the readme file.

**Comment by DwipshikhaLodh** [comment]
Thank you 😊.. okay I'll make those changes

File: C_CPP/Algorithmic Approaches/Dynamic/BellmanFordAlgorithm.md (hunk 1 above)
Comment (DwipshikhaLodh) [reply C_CPP/Algorithmic Approaches/Dynamic/BellmanFordAlgorithm.md:3]: Okay

**Comment by DwipshikhaLodh** [comment]
@kashishahuja2002 I did the changes but here it shows that in readme there's a conflict. How can I resolve it or you will resolve the conflict? Moreover on my local machine there's no file of TSP named which got merged few days back, it's there in this main repo and may be for that reason the conflict is showing.

File: C_CPP/Algorithmic Approaches/Dynamic/BellmanFordAlgorithm.md [hunk 2]
```diff
@@ -0,0 +1,559 @@
+# Bellman Ford Algorithm
+
+
```
Comment (kashishahuja2002) [inline C_CPP/Algorithmic Approaches/Dynamic/BellmanFordAlgorithm.md:3]: Remove these lines.

**Review by kashishahuja2002 — APPROVED** [review]
LGTM

**Comment by kashishahuja2002** [comment]
> @kashishahuja2002 I did the changes but here it shows that in readme there's a conflict. How can I resolve it or you will resolve the conflict? Moreover on my local machine there's no file of TSP named which got merged few days back, it's there in this main repo and may be for that reason the conflict is showing.

You need to fetch upstream and then git pull. then you'll be able to see the file. And for the readme conflict, I'll have to see that. Can you share ss please?

File: C_CPP/Algorithmic Approaches/Dynamic/BellmanFordAlgorithm.md (hunk 2 above)
Comment (DwipshikhaLodh) [reply C_CPP/Algorithmic Approaches/Dynamic/BellmanFordAlgorithm.md:3]: Line 2 & 3?

Comment (DwipshikhaLodh) [reply C_CPP/Algorithmic Approaches/Dynamic/BellmanFordAlgorithm.md:3]: Okay

**Comment by DwipshikhaLodh** [comment]
Ohh yes.. actually instead of doing fetch upstream on C_CPP, I was doing that on main so it was showing that forked repo was up to date. I did upstream so my forked repo got updated for C_CPP branch but the repo on my local machine didn't got updated... Actually may be that's because after my last pr I cloned the repo again and forgot to set the upstream

**Comment by DwipshikhaLodh** [comment]
I updated my local repo with TSP file but how can I update my this branch? Or should I delete this pr and can create another one with another branch?

**Comment by DwipshikhaLodh** [comment]
Screenshot_20211102_142456
This is where conflict is showing because my cpp/dp/bellmanford branch doesn't have the TSP file and even on readme file there's no link

**Comment by DwipshikhaLodh** [comment]
@kashishahuja2002 I created pr of this current branch and pushed that before doing fetch upstream

**Comment by Saviour1001** [comment]
@DwipshikhaLodh when multiple people work on the same file, merge conflicts are bound to happen. I resolved the changes by adding both your changes as well as the already present ones, so now it's sorted 🥳 
Great job on the content !! 🤩

**Comment by DwipshikhaLodh** [comment]
@Saviour1001 Thank you so much ✨ and Thank you @kashishahuja2002 for your time✨

**Pull request merged**